src/bp/core/repositories/events.ts

```diff
@@ -0,0 +1,64 @@
+import * as sdk from 'botpress/sdk'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+
+import Database from '../database'
+import { TYPES } from '../types'
+
+export interface EventRepository {
+  findEvents(fields: Partial<sdk.IO.StoredEvent>, searchParams: sdk.EventSearchParams)
+  deleteBeforeDate(date: Date): Promise<void>
+}
+
+export const DefaultSearchParams: sdk.EventSearchParams = {
+  sortOrder: [{ column: 'createdOn' }],
+  from: 0,
+  count: 10
+}
+
+const UNLIMITED_ELEMENTS = -1
+
+@injectable()
+export class KnexEventRepository implements EventRepository {
+  private readonly TABLE_NAME = 'events'
+
+  constructor(@inject(TYPES.Database) private database: Database) {}
+
+  async findEvents(
+    fields: Partial<sdk.IO.StoredEvent>,
+    params: sdk.EventSearchParams = DefaultSearchParams
+  ): Promise<sdk.IO.StoredEvent[]> {
```

Comment:
do you validate that each fields is a valid field ? otherwise the sql query will crash